e2e/vmss.go

```diff
@@ -82,27 +86,47 @@ func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName str
 	createVMSSCtx, cancel := context.WithTimeout(ctx, createVMSSPollingTimeout)
 	defer cancel()
 
-	pollerResp, err := opts.cloud.vmssClient.BeginCreateOrUpdate(
-		createVMSSCtx,
-		*opts.clusterConfig.cluster.Properties.NodeResourceGroup,
-		vmssName,
-		model,
-		nil,
-	)
-	if err != nil {
-		return nil, err
+	var requestError azure.RequestError
+	var pollerResp *runtime.Poller[armcompute.VirtualMachineScaleSetsClientCreateOrUpdateResponse]
+	for i := 0; i < maxPollingAttempts; i++ {
```

Comment:
can we generalize this flow and put it into a new poller?

checkout pollers.go

Reply:
maybe something like `pollCreateVMSS` or similar